tests/native_api_tests/test_api_model.cpp

```diff
@@ -351,37 +375,183 @@ TEST_CASE("API Model") {
+
+        // update dataset
+        std::vector<Idx> input_source_indptr{0, 1, 1};
+        std::vector<ID> const update_source_id{1};
+        std::vector<double> const update_source_u_ref{0.5};
+        Buffer update_source_buffer{PGM_def_update_source, 1};
+        update_source_buffer.set_nan();
+        update_source_buffer.set_value(PGM_def_update_source_id, update_source_id.data(), -1);
+        update_source_buffer.set_value(PGM_def_update_source_u_ref, update_source_u_ref.data(), -1);
+
+        std::vector<Idx> input_sym_load_indptr{0, 1, 2};
+        std::vector<ID> const update_sym_load_id{2, 5};
+        std::vector<double> const update_sym_load_q_specified{100.0, 300.0};
+        Buffer update_sym_load_buffer{PGM_def_update_sym_load, 2};
+        update_sym_load_buffer.set_nan();
+        update_sym_load_buffer.set_value(PGM_def_update_sym_load_id, update_sym_load_id.data(), -1);
+        update_sym_load_buffer.set_value(PGM_def_update_sym_load_q_specified, update_sym_load_q_specified.data(), -1);
+
+        // update dataset - row
+        DatasetConst update_dataset_row{"update", 1, 2};
+        update_dataset_row.add_buffer("source", -1, 1, input_source_indptr.data(), update_source_buffer);
+        update_dataset_row.add_buffer("sym_load", -1, 2, input_sym_load_indptr.data(), update_sym_load_buffer);
+
+        // update dataset - col
+        DatasetConst update_dataset_col{"update", 1, 2};
+        DatasetConst update_dataset_col_no_id{"update", 1, 2};
+
+        update_dataset_col.add_buffer("source", -1, 1, input_source_indptr.data(), nullptr);
+        update_dataset_col.add_attribute_buffer("source", "id", update_source_id.data());
+        update_dataset_col.add_attribute_buffer("source", "u_ref", update_source_u_ref.data());
+
+        update_dataset_col.add_buffer("sym_load", -1, 2, input_sym_load_indptr.data(), nullptr);
+        update_dataset_col.add_attribute_buffer("sym_load", "id", update_sym_load_id.data());
+        update_dataset_col.add_attribute_buffer("sym_load", "q_specified", update_sym_load_q_specified.data());
+
+        update_dataset_col_no_id.add_buffer("source", -1, 1, input_source_indptr.data(), nullptr);
+        update_dataset_col_no_id.add_attribute_buffer("source", "u_ref", update_source_u_ref.data());
+
+        update_dataset_col_no_id.add_buffer("sym_load", -1, 2, input_sym_load_indptr.data(), nullptr);
+        update_dataset_col_no_id.add_attribute_buffer("sym_load", "q_specified", update_sym_load_q_specified.data());
+
+        // output data
+        Buffer batch_node_output{PGM_def_sym_output_node, 2};
+        batch_node_output.set_nan();
+        DatasetMutable batch_output_dataset{"sym_output", 1, 2};
+        batch_output_dataset.add_buffer("node", 1, 2, nullptr, batch_node_output);
+
+        // options
+        Options const batch_options{};
+
+        SUBCASE("Row-based input dataset") {
+            Model row_model{50.0, input_dataset_row};
```

Comment:
just `model` suffices because it's in its own scope/subcase. feel free to leave as is, though

Reply:
shadowing